docs/cookbook/server-managed-meta-tags.md

```diff
@@ -0,0 +1,351 @@
+
+          if (httpEquiv) {
+            attributes['http-equiv'] = httpEquiv;
+          }
+
+          let content = null;
+          if (innerContent != null) {
+            content = typeof innerContent === 'string'
+              ? innerContent
+              : JSON.stringify(innerContent);
+          }
+
+          return h(tagName, attributes, content);
+        })
+      );
+    };
+  }
+};
+</script>
+```
+
+== React
+
+```jsx
+import React from 'react';
+import { Head, usePage } from '@inertiajs/react';
+
+const MetaTags = () => {
+  const { _inertia_meta: meta } = usePage().props;
+  return (
+    <Head>
+      {meta.map((meta) => {
+        const { tagName, innerContent, headKey, httpEquiv, ...attrs } = meta;
+
+        let stringifiedInnerContent;
+        if (innerContent != null) {
+          stringifiedInnerContent =
+            typeof innerContent === 'string' ? innerContent : JSON.stringify(innerContent);
+        }
+
+        return React.createElement(tagName, {
+          key: headKey,
+          'head-key': headKey,
+          ...(httpEquiv ? { 'http-equiv': httpEquiv } : {}),
+          ...attrs,
+          ...(stringifiedInnerContent
+            ? { dangerouslySetInnerHTML: { __html: stringifiedInnerContent } }
+            : {}),
+        });
+      })}
+    </Head>
+  );
+};
+
+export default MetaTags;
+```
+
+== Svelte 4|Svelte 5
+
+```svelte
+<!-- MetaTags.svelte -->
+<script>
+  import { onMount } from 'svelte';
+  import { page } from '@inertiajs/svelte';
+
+  $: metaTags = ($page.props._inertia_meta ?? []).map(({
+    tagName,
+    headKey,
+    innerContent,
+    httpEquiv,
+    ...attrs
+  }) => ({
+    tagName,
+    headKey,
+    innerContent,
+    attrs: httpEquiv ? { ...attrs, 'http-equiv': httpEquiv } : attrs
+  }));
+
+  // Svelte throws warnings if we render void elements like meta with content
+  $: voidTags = metaTags.filter(tag => tag.innerContent == null);
+  $: contentTags = metaTags.filter(tag => tag.innerContent != null);
+
+  let ready = false;
+
+  onMount(() => {
+    // Clean up server-rendered tags
+    document.head
+      .querySelectorAll('[inertia]')
+      .forEach(el => el.remove());
+
+    ready = true;
+  });
+</script>
+
+<svelte:head>
+  {#if ready}
+    <!-- Void elements (no content) -->
+    {#each voidTags as tag (tag.headKey)}
+      <svelte:element
+        this={tag.tagName}
+        inertia={tag.headKey}
+        {...tag.attrs}
+      />
+    {/each}
+
+    <!-- Elements with content -->
+    {#each contentTags as tag (tag.headKey)}
+      <svelte:element
+        this={tag.tagName}
+        inertia={tag.headKey}
+        {...tag.attrs}
+      >
+        {@html
+          typeof tag.innerContent === 'string'
+            ? tag.innerContent
+            : JSON.stringify(tag.innerContent)
+        }
+      </svelte:element>
+    {/each}
+  {/if}
+</svelte:head>
+
+```
+:::
+
+## Rendering Meta Tags
+
+Tags are defined as plain hashes and conform to the following structure:
```

Comment:
I looked at the Rails meta_tags gem and Remix.run's meta objects for inspiration. I decided to minimize "magic". The data structure just describes an HTML, with a few exceptions:

* `tag_name` will default to `:meta` if it isn't defined
* `inner_content` is used for non-void tags
* `title` has shortcut syntax